### PR TITLE
[PWX-35030]: T34542906: Newly set pool labels should persist post pool expand using add-disk

### PR DESCRIPTION
**Pull request by ragsingh-px**
**What this PR does / why we need it**:
Automation test for Newly set pool labels should persist post pool expand using add-disk
**Which issue(s) this PR fixes** (optional)
Closes # PWX-35030

**Special notes for your reviewer**:
Testing link:- https://jenkins.pwx.dev.purestorage.com/user/ragsingh@purestorage.com/my-views/view/all/job/Users/job/Raghvendra/job/raghav_pxdv-7-dev/63

### CODE REVIEW

File: tests/basic/pool_function_test.go [hunk 1]
```diff
@@ -932,3 +934,106 @@ var _ = Describe("{PoolExpandTestLimits}", func() {
...
+			log.FailOnError(err, "Failed to update the label on the pool %s", poolIDToResize)
+		})
+
+		stepLog = "Move pool to maintenance mode"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			log.InfoD(fmt.Sprintf("Entering pool maintenance mode on node %s", storageNode.Name))
+			err = Inst().V.EnterPoolMaintenance(*storageNode)
+			log.FailOnError(err, fmt.Sprintf("fail to enter node %s in maintenance mode", storageNode.Name))
+			status, _ := Inst().V.GetNodeStatus(*storageNode)
```
[the remark this answers is not in the document]
Comment (vinayakshnd) [reply tests/basic/pool_function_test.go:1049]: @ragsingh-px Lets use `WaitForPoolStatusToUpdate` API to wait till PX comes up in maintenance mode.

For eg. as used here: https://github.com/portworx/torpedo/blob/master/tests/common.go#L7564

File: tests/basic/pool_function_test.go [hunk 2]
```diff
@@ -994,3 +995,107 @@ var _ = Describe("{PoolExpandTestLimits}", func() {
...
+			dash.VerifyFatal(err, nil, "pool expansion requested successfully")
+			resizeErr := waitForOngoingPoolExpansionToComplete(poolIDToResize)
+			dash.VerifyFatal(resizeErr, nil, "Pool expansion does not result in error")
+
+		})
+
+		stepLog = "Exit pool out of maintenance mode"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			t := func() (interface{}, bool, error) {
```
Comment (vinayakshnd) [inline tests/basic/pool_function_test.go:1073]: Lets reduce this code by just using `WaitForPoolStatusToUpdate` as used here: https://github.com/portworx/torpedo/blob/master/tests/common.go#L7587C8-L7587C33

File: tests/basic/pool_function_test.go [hunk 3]
```diff
@@ -994,3 +995,107 @@ var _ = Describe("{PoolExpandTestLimits}", func() {
...
+
+var _ = Describe("{CheckPoolLabelsAfterAddDisk}", func() {
+
+	var testrailID = 34542906
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/tests/view/34542906
+
+	BeforeEach(func() {
+		StartTorpedoTest("CheckPoolLabelsAfterAddDisk",
+			"Initiate pool expansion and Newly set pool labels should persist post pool expand add-disk operation", nil, testrailID)
+		contexts = scheduleApps()
```
Comment (vinayakshnd) [inline tests/basic/pool_function_test.go:1007]: After this command, Lets validate that scheduled Apps come up successfully using `ValidateApplications` API.
https://github.com/portworx/torpedo/blob/master/tests/common.go#L1891

Comment (ragsingh-px) [reply tests/basic/pool_function_test.go:1007]: Is that necessary? We are not doing that anywhere else.

File: tests/basic/pool_function_test.go (hunk 2 above)
Comment (ragsingh-px) [reply tests/basic/pool_function_test.go]: fixed

File: tests/basic/pool_function_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ragsingh-px) [reply tests/basic/pool_function_test.go:1123]: fixed

**Review by vinayakshnd — APPROVED** [review]
LGTM

**Pull request merged**